Logging/LogHandler/FileLogHandler.cs

```diff
@@ -10,12 +9,30 @@ namespace Anvil.CSharp.Logging
     /// </summary>
     public class FileLogHandler : AbstractAnvilBase, ILogHandler
     {
+        /// <summary>
+        /// The type of log files the handler should generate.
+        /// </summary>
+        public enum LogType
+        {
+            /// <summary>
+            /// New logs are appended to the existing log file.
+            /// </summary>
+            Append,
+            /// <summary>
+            /// A new log file is created each session, overwriting the previous one if it exists.
+            /// </summary>
+            Replace,
+        }
+
         public const string LOG_CONTEXT_CALLER_DERIVED_TYPE = "{0}";
         public const string LOG_CONTEXT_CALLER_FILE = "{1}";
         public const string LOG_CONTEXT_CALLER_METHOD = "{2}";
         public const string LOG_CONTEXT_CALLER_LINE = "{3}";
 
-        private readonly StreamWriter m_Writer;
+        private StreamWriter m_Writer;
+        private readonly string m_Path;
+        private readonly LogType m_LogType;
+        private readonly bool m_Rotate;
```

Comment:
Minor but since we're using an enum for `LogType` with two values, should keep the consistency and have an enum for `RotateBehaviour` instead of a bool?

Reply:
My justification was that "Replace" isn't the obvious opposite of "Append", they're two distinct settings to me. Rotation though is just a switch - it's on or off. The enum would be `Rotate` and ... `DontRotate`?

Reply:
Well come on, we're professionals, it would be `DO_NOT_ROTATE` obvi. /s

That's a fair point. I guess in my mind I had thought it would have been `Rotate` and `Replace` but that's confusing with `Replace` already being used in the `LogType`. 

Oh well, it was minor anyway, I'm cool with the bool

Reply:
The best I can come up with is `Default`, which doesn't tell you anything. It's not that the logs would be replaced without rotation, it's that they would just grow indefinitely as logs come in, and the best way I can think to represent that is `shouldRotate: false`.

I will keep the bool until a stronger opinion appears!